### PR TITLE
Add pull/push workflow for translations

### LINKED ISSUE #104: [2pt] Create GitHub Actions workflows to translate documentation
The documentation translation process should be automated. We need to implement a set of tasks at each stage of tech writer's workflow to maintain 100% translation for each repository containing documentation.

- Tech writer creates PR with a documentation update (rst sources).
  - (push-translation on pull_request) GitHub workflow builds _po_ files from _rst_ and sends them to the same name branch in Crowdin
- Pull translations back
  - (pull-translation on workflow_dispatch) When translation of the corresponding changes is ready, tech writer manually triggers GitHub workflow for the translated branch (PR branch)
- Tech writer or someone responsible merges PR
  - (upload-translation on push to master) GithHub workflow upload merged translations from _po_ files into Crowdin master (or main) branch

### PR DESCRIPTION
**Pull request by artembo**
Documentation sources are .rst files where a tech writer or a developer writes documentation. As soon as we have two languages supported on our website we need to maintain translated 100% of the documentation. For translation, we use Crowdin - saas solution for translating any texts.  
While a new PR with changes in .rst is opened, the translations for these changes are preparing by translators in Crowdin.  
When the translations are ready we can run __Pull translations__ job to download and commit translations into the PR branch.  
This diagram shows how the process works:
crowdin-time-diagram

- add cleanup.py for re-arranging .po files and initial re-arranged .po files
- push-translation.yml push translation sources to Crowdin on pull request when .rst is changed, create a deployment with a link to Crowdin translations
- pull-translation.yml for getting translations from Crowdin on demand when translation is ready and auto-commit them into the same PR branch
- upload-translations.yml for upload translations to the master Crowdin branch from the git master branch
- destroy deployment with a link to Crowdin translations when the translations (in PR) are merged

Closes #104

### CODE REVIEW

File: .github/workflows/push-translation.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: Push translation sources
+
+on:
+  pull_request:
+    paths:
+      - 'doc/**/*'
```
Comment (NickVolynkin) [inline .github/workflows/push-translation.yml:6]:
```suggestion
      - 'doc/**/*.rst'
      - 'doc/locale/**/*.po'
      - '.github/workflows/upload-translations.yml'
```

Comment (NickVolynkin) [inline .github/workflows/push-translation.yml:6]:
```suggestion
      - 'doc/**/*.rst'
      - 'doc/conf.py'
      - '.github/workflows/push-translation.yml'
```

File: .github/workflows/upload-translations.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+name: Update translations on the main branch
+
+on:
+  push:
+    paths:
+      - 'doc/**/*'
```
Comment (NickVolynkin) [inline .github/workflows/upload-translations.yml:6]:
```suggestion
      - 'doc/**/*.rst'
      - 'doc/locale/**/*.po'
      - '.github/workflows/upload-translations.yml'
```

File: .gitignore [hunk 3]
```diff
@@ -85,3 +85,5 @@ deploy/*
 
 ci/helm-chart/templates/crds/tarantool.io_*s_crd.yaml
 ci/helm-chart/templates/crds/tarantool_*_cr.yaml
+
+doc/locale/en/
```
Comment (NickVolynkin) [inline .gitignore:89]:
```suggestion
doc/locale/en/
doc/output/
```

**Review by NickVolynkin — APPROVED** [review]

File: .github/workflows/upload-translations.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
+name: Update translations on the main branch
+
+on:
+  push:
+    paths:
+      - 'doc/**/*.rst'
+      - 'doc/locale/**/*.po'
```
Comment (vanyarock01) [inline .github/workflows/upload-translations.yml:7]: Why do I need to run a pull when this file changes?

File: doc/locale/ru/LC_MESSAGES/cartridge_kubernetes_guide/index.po [hunk 5]
```diff
@@ -0,0 +1,807 @@
+
+msgid "Tarantool Cartridge on Kubernetes"
```
Comment (vanyarock01) [inline doc/locale/ru/LC_MESSAGES/cartridge_kubernetes_guide/index.po:2]: I'm confused. Shouldn't the translation result be in the .po files?

**Comment by vanyarock01** [comment]
How to understand that the translation is ready?

File: .github/workflows/upload-translations.yml (hunk 4 above)
Comment (NickVolynkin) [reply .github/workflows/upload-translations.yml:7]: We merge translation files from a PR and then re-upload translations back to Crowdin

**Comment by NickVolynkin** [comment]
> How to understand that the translation is ready?

We see it at Crowdin. For example, at this moment it isn't ready at all:

Screenshot 2021-07-23 at 15 17 46

File: doc/locale/ru/LC_MESSAGES/cartridge_kubernetes_guide/index.po (hunk 5 above)
Comment (NickVolynkin) [reply doc/locale/ru/LC_MESSAGES/cartridge_kubernetes_guide/index.po:2]: Translation can be in .po files and only there. But translation is done by humans, and they couldn't do it yet, and won't be until this PR gets merged.

**Review by vanyarock01 — APPROVED** [review]

**Pull request merged**